overlord.pl

```diff
@@ -371,6 +384,14 @@ sub squidHasSwapouts
     return $mgrPage =~ /SWAPOUT_WRITING/;
 }
 
+# whether a job with a matching type still runs
+sub hasJob
+{
+    my $jobType = shift;
+    my $mgrPage = &getCacheManagerResponse('jobs')->{content};
+    return $mgrPage =~ /$jobType/m;
```

Comment:
Please tighten this match to look just at the reported AsyncJob::typeName field _values_, not the entire response content. For example, if a `/type|rock/` regex should not match the following mgr:jobs response, but an `/ufs|updater/i` regex should match it:

```
job1087:
    type: 'UFS::HeaderWriter'
job1089:
    type: 'Rock::HeaderUpdater'
```

>  we match the entire regex passed from Daft. Is this enough 'tightening'? Or we need a multiline match of two relevant of strings

Neither, AFAICT. You are probably thinking about the regex. This change request is about the content (areas) that the regex is applied to (job type values specifically). Those values are not multiline.  In the above example, the values to check are `UFS::HeaderWriter` and `Rock::HeaderUpdater`. Matching one of the values is enough.

Reply:
Done at 5330bd.